### PR TITLE
[2.1] Improve smileys popup behavior

### PR DESCRIPTION
**Pull request by DiegoAndresCortes**
- Now more consistent positioning
- Added closing when clicking/tapping outside

It now stays in place, even without smileys present on the editor

Screenshot_103

### CODE REVIEW

File: Themes/default/scripts/jquery.sceditor.smf.js [hunk 1]
```diff
@@ -145,7 +152,7 @@
 						adjheight = closeButton.height() + titlebar.height();
 						$dropdown = $('<div class="centerbox sceditor-smileyPopup">')
 							.append(popupContent)
-							.appendTo($('.sceditor-container'));
+							.appendTo($('center'));
```
Comment (live627) [inline Themes/default/scripts/jquery.sceditor.smf.js]: this a tag selector? What happens if more than one exist?

Comment (DiegoAndresCortes) [reply Themes/default/scripts/jquery.sceditor.smf.js]: I guess
`.sceditor-insertemoticon center[unselectable="on"]` would be better?

Comment (live627) [reply Themes/default/scripts/jquery.sceditor.smf.js]: don't need the attribute selector, the more specific selector will only ever return one element

```suggestion
							.appendTo($('.sceditor-insertemoticon center'));
```

Comment (DiegoAndresCortes) [reply Themes/default/scripts/jquery.sceditor.smf.js]: Cool, will update this weekend with some changes.

**Comment by live627** [comment]
Can you rebase this now that https://github.com/SimpleMachines/SMF/pull/7971 has been merged? This now has merge conflicts.

**Comment by DiegoAndresCortes** [comment]
Done

**Comment by Sesquipedalian** [comment]
How does this look to you, @live627?

File: Themes/default/scripts/jquery.sceditor.smf.js [hunk 2]
```diff
@@ -126,6 +126,13 @@
 						closeButton = $('<span class="button">').text(base._('Close')).click(function () {
 							$(base.editorMainWrapper.querySelector(".sceditor-smileyPopup")).fadeOut('fast');
 						});
+						$(document).mouseup(function (e) {
+							if (allowHide && !popupContent.is(e.target) && popupContent.has(e.target).length === 0)
+								$(".sceditor-smileyPopup").fadeOut('fast');
+						}).keyup(function (e) {
+							if (e.keyCode === 27)
+								$(".sceditor-smileyPopup").fadeOut('fast');
+						});
```
Comment (live627) [inline Themes/default/scripts/jquery.sceditor.smf.js]:
```suggestion
						$(document).mouseup(function (e) {
							if (allowHide && !popupContent.is(e.target) && popupContent.has(e.target).length === 0)
								$(smileyPopup).fadeOut('fast');
						}).keyup(function (e) {
							if (e.keyCode === 27)
								$(smileyPopup).fadeOut('fast');
						});
```

**Pull request merged**